tests/integration/utils/utils.go

```diff
@@ -0,0 +1,205 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	cmtabcitypes "github.com/cometbft/cometbft/abci/types"
+	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
+	dbm "github.com/cosmos/cosmos-db"
+
+	"cosmossdk.io/core/address"
+	"cosmossdk.io/core/appmodule"
+	"cosmossdk.io/log"
+	"cosmossdk.io/store"
+	"cosmossdk.io/store/metrics"
+	storetypes "cosmossdk.io/store/types"
+	authtx "cosmossdk.io/x/auth/tx"
+	authtypes "cosmossdk.io/x/auth/types"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/runtime"
+	testutil "github.com/cosmos/cosmos-sdk/testutil/integration"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	consensusparamkeeper "github.com/cosmos/cosmos-sdk/x/consensus/keeper"
+	consensusparamtypes "github.com/cosmos/cosmos-sdk/x/consensus/types"
+	simtestutil "github.com/cosmos/cosmos-sdk/x/simulation/helper"
+)
+
+const appName = "integration-app"
+
+// App is a test application that can be used to test the integration of modules.
+type App struct {
+	*baseapp.BaseApp
+
+	ctx           sdk.Context
+	logger        log.Logger
+	moduleManager module.Manager
+	queryHelper   *baseapp.QueryServiceTestHelper
+}
+
+// Can move to itegration
+
+// NewIntegrationApp creates an application for testing purposes. This application
+// is able to route messages to their respective handlers.
+func NewIntegrationApp(
```

Comment:
we should check if users are using this?

Reply:
Its only used by integration test

Reply:
Yes, but we expect users to use `NewIntegrationApp` to create their integration tests.
Otherwise they need to copy/paste this (you shouldn't and cannot import /tests)

Reply:
Moved integration out to `testutil/integration`, not related to auth or bank anymore. Will mark this cmt as resolved. Thanks u